src/System.Text.Json/src/System/Text/Json/Serialization/WriteStackFrame.cs

```diff
@@ -35,41 +44,102 @@ internal void Initialize(Type type, JsonSerializerOptions options)
             }
         }
 
-        internal void Reset()
+        public void WriteObjectOrArrayStart(ClassType classType, Utf8JsonWriter writer, bool writeNull = false)
```

Comment:
Change to separate method rather than one with optional argument for `writeNull` to avoid bloat in `WriteObjectOrArrayStart`?

Reply:
Keeping it this way saves ~50LOC